packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/layoutdata/HandleLayoutData.java

```diff
@@ -0,0 +1,21 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.layoutdata;
+
+/**
+ * The layout data of a node.
+ *
+ * @author mcharfadi
+ */
+public record HandleLayoutData(String edgeId, Position position, String handlePosition, String type) {
```

Comment:
Are we allowing nullable data in other layout data?

Reply:
It seems that we are, we may change that one day but it's not relevant for this PR